agent/provider/src/cli/keystore.rs

```diff
@@ -131,44 +152,116 @@ fn remove(config: ProviderConfig, remove: Remove) -> anyhow::Result<()> {
 }
 
 fn print_cert_list(config: &ProviderConfig, certs_data: Vec<Cert>) -> anyhow::Result<()> {
-    let mut table = CertTable::new();
+    let mut table_builder = CertTableBuilder::new();
     for data in certs_data {
-        table.add(data);
+        table_builder.add(data);
     }
-    table.print(config)?;
+
+    table_builder.build().print(config)?;
     Ok(())
 }
 
-struct CertTable {
-    table: ResponseTable,
+fn find_ids_by_prefix(certs: &[Cert], prefix: &str) -> Vec<String> {
+    certs
+        .iter()
+        .map(|cert| cert.id())
+        .filter(|id| id.starts_with(prefix))
```

Comment:
We could build once a https://docs.rs/trie-rs/latest/trie_rs from `certs` and use it for prefix search, but it would be an overkill.

Reply:
Agreed, I think it's better to leave it simple